### PR TITLE
Remove TTL, use PoW instead

### PR DESCRIPTION
**Pull request by EVODelavega**
Reverts changes for #7969 to use PoW instead. Includes a minor change to validator commands in check & deliver Tx (skips the block height check)

### CODE REVIEW

File: core/blockchain/abci/tm_node.go [hunk 1]
```diff
@@ -143,8 +143,6 @@ func overwriteConfig(config *config.Config) {
 	config.Consensus.CreateEmptyBlocks = true
 	// enforce using priority mempool
 	config.Mempool.Version = "v1"
-	// ensure rechecking tx is enabled
-	config.Mempool.Recheck = true
```
Comment (wwestgarth) [inline core/blockchain/abci/tm_node.go:147]: Do we not still need this so that the checks in the PoW engine happen after every block? @ze97286

Comment (jeremyletang) [reply core/blockchain/abci/tm_node.go:147]: yea we do.

**Review by davidsiska-vega — DISMISSED** [review]
Looked at `defaults.go` and `keys.go`.

**Review by davidsiska-vega — DISMISSED** [review]
Just default and keys.

File: core/blockchain/abci/tm_node.go (hunk 1 above)
Comment (EVODelavega) [reply core/blockchain/abci/tm_node.go:147]: Yes, got removed in the revert, and I've added it back in the last commit I pushed :)

**Review by davidsiska-vega — DISMISSED** [review]

File: core/pow/engine.go [hunk 2]
```diff
@@ -380,6 +380,17 @@ func (e *Engine) verify(tx abci.Tx) (byte, error) {
...
 	txHash := hex.EncodeToString(tx.Hash())
 
+	// check for replay attacks
+	if _, ok := e.seenTx[txHash]; ok {
+		e.log.Error("replay attack: txHash already used", logging.String("tx-hash", txHash), logging.String("tid", tx.GetPoWTID()), logging.String("party", tx.Party()), logging.String("command", tx.Command().String()))
+		return h, errors.New("transaction hash already used")
+	}
+
+	// validator commands skip PoW verification
+	if tx.Command().IsValidatorCommand() {
```
Comment (ze97286) [inline core/pow/engine.go:390]: just noting here that it is a change of behaviour which means validators block is not check. I don't feel strongly against it but I don't think it's necessary either.

**Review by jeremyletang — DISMISSED** [review]

**Review by jeremyletang — APPROVED** [review]

**Review by davidsiska-vega — APPROVED** [review]

File: core/pow/engine.go (hunk 2 above)
Comment (EVODelavega) [reply core/pow/engine.go:390]: Agreed, after looking at the [spec update PR](https://github.com/vegaprotocol/specs/pull/1673), I checked with @davidsiska-vega to make sure that everyone is aware of this change, and it's the intended behaviour.

**Review by ze97286 — APPROVED** [review]

**Review by gordsport — APPROVED** [review]

**Pull request merged**